chain/network/src/test_utils.rs

```diff
@@ -25,22 +25,22 @@ use tracing::debug;
 static OPENED_PORTS: Lazy<Mutex<HashSet<u16>>> = Lazy::new(|| Mutex::new(HashSet::new()));
 
 /// Returns available port.
-pub fn open_port() -> u16 {
+pub fn open_port() -> std::net::SocketAddr {
```

Comment:
This still suffers from TOCTOU. Is there any reason this can’t return the newly added `ListenerAddr`?

Reply:
also we have tests which use the json-representation of the config, in which case the ListenerAddr representation of the address will be dropped. However I guess I could replace ListenerAddr with a global set (just like open_ports does) of guards for tcp ports allocated during the test. WDYT?

Reply:
I think its better to land as is right now and look into addressing it as a follow-up if you think it is a good use of our current resources.